conduit/pipeline_test.go

```diff
@@ -422,3 +433,123 @@ func TestPipelineErrors(t *testing.T) {
 	assert.Error(t, pImpl.Error(), fmt.Errorf("exporter"))
 
 }
+
+// TestBlockMetaDataFile tests that metadata.json file is created as expected
+func TestBlockMetaDataFile(t *testing.T) {
+
+	var pImporter importers.Importer = &mockImporter{}
+	var pProcessor processors.Processor = &mockProcessor{}
+	var pExporter exporters.Exporter = &mockExporter{}
+
+	datadir := t.TempDir()
+	pImpl := pipelineImpl{
+		cfg: &PipelineConfig{
+			ConduitConfig: &Config{
+				Flags:          nil,
+				ConduitDataDir: datadir,
+			},
+			Importer: NameConfigPair{
+				Name:   "",
+				Config: map[string]interface{}{},
+			},
+			Processors: []NameConfigPair{
+				{
+					Name:   "",
+					Config: map[string]interface{}{},
+				},
+			},
+			Exporter: NameConfigPair{
+				Name:   "",
+				Config: map[string]interface{}{},
+			},
+		},
```

Comment:
Minor: It's potentially worth adding this to a test util since I see us using an empty config in multiple places.

Reply:
I'm leaving this as is because moving PipelineConfig to test util creates a dependency cycle.